src/Components/Server/src/Circuits/CircuitHost.cs

```diff
@@ -48,7 +48,7 @@ internal class CircuitHost : IAsyncDisposable
             if (CircuitId.Secret is null)
             {
                 // Prevent the use of a 'default' secret.
-                throw new ArgumentException(nameof(circuitId));
+                throw new ArgumentNullException(nameof(circuitId));
```

Comment:
Is this really correct? It looks as if `circuitId` must *not* be null here otherwise evaluating `CircuitId.Secret` would already have thrown above.

Reply:
```suggestion
                throw new ArgumentException($"Property '{nameof(CircuitId.Secret)}' cannot be null.", nameof(circuitId));
```